src/GraphQL/Types/Scalars/EnumerationGraphType.cs

```diff
@@ -62,39 +62,61 @@ public void AddValue(EnumValueDefinition value)
         public EnumValues Values { get; }
 
         /// <inheritdoc/>
-        public override object Serialize(object value)
+        public override object ParseLiteral(IValue value) => value switch
         {
-            var valueString = value.ToString(); //TODO: find only by value?
-            var foundByName = Values.FindByName(valueString);
-            if (foundByName != null)
-            {
-                return foundByName.Name;
-            }
+            EnumValue enumValue => Values.FindByName(enumValue.Name)?.Value ?? ThrowLiteralConversionError(value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
-            var foundByValue = Values.FindByValue(value);
-            return foundByValue?.Name;
-        }
+        /// <inheritdoc/>
+        public override bool CanParseLiteral(IValue value) => value switch
+        {
+            EnumValue enumValue => Values.FindByName(enumValue.Name) != null,
+            NullValue _ => true,
+            _ => false
+        };
+
+        /// <inheritdoc/>
+        public override object ParseValue(object value) => value switch
+        {
+            string s => Values.FindByName(s)?.Value ?? ThrowValueConversionError(value),
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value) => value is EnumValue enumValue ? ParseValue(enumValue.Name) : null;
+        public override bool CanParseValue(object value) => value switch
+        {
+            string s => Values.FindByName(s) != null,
+            null => true,
+            _ => false
+        };
 
         /// <inheritdoc/>
-        public override object ParseValue(object value)
+        public override object Serialize(object value)
         {
             if (value == null)
-            {
                 return null;
-            }
 
-            var found = Values.FindByName(value.ToString());
-            return found?.Value;
+            var foundByValue = Values.FindByValue(value);
+            if (foundByValue == null)
+                ThrowSerializationError(value);
+
+            return foundByValue.Name;
```

Comment:
```suggestion
            var foundByValue = Values.FindByValue(value);
            return foundByValue == null
                ? ThrowSerializationError(value)
                : foundByValue.Name;
```

Also remove `FindByName`. I was used only here and it is `internal`.

Reply:
🤦 I see other usages here.